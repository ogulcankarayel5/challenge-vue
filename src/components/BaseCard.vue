<template>
  <div class="baseCard" v-bind="$attrs">
    <slot />
  </div>
</template>

<script>
export default {
  name: "BaseCard",
};
</script>

<style scoped lang="sass">
.baseCard
    width: 440px
    height: auto
    padding: 15px
    margin-top: 30px
    background-color: transparent
    display: flex
    justify-content: center

    cursor: pointer
</style>
