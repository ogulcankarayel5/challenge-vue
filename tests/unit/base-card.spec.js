import { mount } from "@vue/test-utils";
import BaseCard from "@/components/BaseCard.vue";

describe("BaseCard.vue", () => {
  it("snapshot", () => {
    const wrapper = mount(BaseCard);

    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders component correctly with slot", () => {
    const wrapper = mount(BaseCard, {
      slots: {
        default: "<div>basecard</div>",
      },
    });
    expect(wrapper.html()).toContain("<div>basecard</div>");
  });

  it("renders component with correct class", () => {
    const wrapper = mount(BaseCard);
    expect(wrapper.find(".baseCard")).toBeTruthy();
  });
});
