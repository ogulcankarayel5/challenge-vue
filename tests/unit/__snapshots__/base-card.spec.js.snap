// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard.vue snapshot 1`] = `
<div
  class="baseCard"
>
  
  
</div>
`;
